### PR TITLE
delete task data after delete task

### PR DESCRIPTION
**Pull request by dmitryagapov**
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat/apps/engine/signals.py [hunk 1]
```diff
@@ -17,3 +28,12 @@ def update_task_status(instance, **kwargs):
         db_task.status = status
         db_task.save()
 
+
+@receiver(post_delete, sender=Task, dispatch_uid="delete_task_files_on_delete_task")
+def delete_task_files_on_delete_task(instance, **kwargs):
+    shutil.rmtree(instance.get_task_dirname, ignore_errors=True)
```
Comment (nmanovic) [inline cvat/apps/engine/signals.py:34]: Looks wrong: instance.get_task_dirname -> instance.get_task_dirname()

Could you please add a test that the bug is fixed?

**Comment by nmanovic** [comment]
Need to add a line into CHANGELOG.md about the fix. Also could you please mention the fixed issue in the PR? It will be closed automatically after it is merged.

**Comment by nmanovic** [comment]
to disable the fp warning please use the advice: https://docs.pylint.org/en/latest/faq.html#do-i-have-to-remember-all-these-numbers

File: cvat/apps/engine/signals.py (hunk 1 above)
Comment (dmitryagapov) [reply cvat/apps/engine/signals.py:34]: > Looks wrong: instance.get_task_dirname -> instance.get_task_dirname() 

Thanks, I'll fix it. By the way all such functions look like properties.

> Could you please add a test that the bug is fixed?
Could you please give an example?

Comment (nmanovic) [reply cvat/apps/engine/signals.py:34]: https://github.com/openvinotoolkit/cvat/tree/develop/cvat/apps/engine/tests

Comment (nmanovic) [reply cvat/apps/engine/signals.py:34]: Properties don't have a verb as a prefix. Usually `task_dirname` is a property and `get_task_dirname` is a function.

**Review by nmanovic — APPROVED** [review]

**Pull request merged**